pkgs/applications/misc/p2pool/default.nix

```diff
@@ -0,0 +1,37 @@
+{ stdenv
+, cmake
+, fetchgit
+, gss
+, hwloc
+, lib
+, libsodium
+, libuv
+, openssl
+, pkg-config
+, zeromq
+}:
+
+stdenv.mkDerivation rec {
+  pname = "p2pool";
+  version = "1.3.1";
+
+  src = fetchgit {
```

Comment:
For gitlab/github there are specialised fetchers that download a tarball instead of cloning the git repository (slower and not always reproducible). So, use `fetchFromGitHub` in this case.